client/app/bundles/course/experience-points/disbursement/pages/DisbursementIndex/index.tsx

```diff
@@ -0,0 +1,176 @@
+import { FC, useEffect, useState } from 'react';
+import {
+  defineMessages,
+  FormattedMessage,
+  injectIntl,
+  WrappedComponentProps,
+} from 'react-intl';
+import { useDispatch, useSelector } from 'react-redux';
+import { toast } from 'react-toastify';
+import { AppDispatch, AppState } from 'types/store';
+import LoadingIndicator from 'lib/components/LoadingIndicator';
+import PageHeader from 'lib/components/pages/PageHeader';
+import { Grid, Paper, Tab, Tabs } from '@mui/material';
+import palette from 'theme/palette';
+import FormatListBulletedIcon from '@mui/icons-material/FormatListBulleted';
+import { Group } from '@mui/icons-material';
+import {
+  getAllCourseGroupMiniEntities,
+  getAllFilteredUserMiniEntities,
+  getAllForumDisbursementUserEntities,
+  getAllForumPostEntities,
+  getCurrentGroup,
+  getFilters,
+} from '../../selectors';
+import DisbursementForm from '../../components/forms/DisbursementForm';
+import { fetchDisbursements, fetchForumDisbursements } from '../../operations';
+import ForumDisbursementForm from '../../components/forms/ForumDisbursementForm';
+import FilterForm from '../../components/forms/FilterForm';
+
+type Props = WrappedComponentProps;
+
+const translations = defineMessages({
+  fetchDisbursementFailure: {
+    id: 'course.experience-points.disbursement.index.fetch.failure',
+    defaultMessage: 'Failed to retrieve Disbursements.',
+  },
+  disbursements: {
+    id: 'course.experience-points.disbursement.index.disbursement',
+    defaultMessage: 'Disburse Experience Points',
+  },
+  forumTab: {
+    id: 'course.experience-points.disbursement.index.forumTab',
+    defaultMessage: 'Forum Participation',
+  },
+  generalTab: {
+    id: 'course.experience-points.disbursement.index.generalTab',
+    defaultMessage: 'General Disbursement',
+  },
+});
+
+const DisbursementIndex: FC<Props> = (props) => {
+  const { intl } = props;
+  const dispatch = useDispatch<AppDispatch>();
+  const [isLoading, setIsLoading] = useState(true);
+  const [tabValue, setTabValue] = useState('forum-disbursement-tab');
+
+  const currentGroup = useSelector((state: AppState) => getCurrentGroup(state));
```

Comment:
Instead of using useSelector here, you can directly call it in the DisbursementForm component so you dont need to pass it down.

i believe currentGroup isnt use anywhere else right?

Similarly for courseGroups, users, filters, forumUsers and forumPosts. If any of those is not really used in this file and it's passed to a child component, it's better to call the useSelector function inside the child component directly.

Reply:
I could move some selectors to their child components, but I do use some of them with useEffect hooks in the child component, for example in the ForumDisbursementForm, the updating of experience point field based on the change in weekly cap value, is necessary for the fields to correctly update. I think the ones that are used multiple times are filters and forumUsers, should I keep these and move the rest?

I mainly decided to put it all in index so that the child components can react based on the changes in redux state, and not just re-render every time redux state change happens

Reply:
Noted